projects/dotcms-field-elements/src/utils/errorHandling.tsx

```diff
@@ -0,0 +1,62 @@
+const fieldValidationMap = {
+    options: stringValidator,
+    regexCheck: regexValidator
+};
+
+interface PropValidationInfo<T> {
+    field: {
+        type: string;
+        name: string;
+    };
+    name: string;
+    value: T;
+}
+
+class DotFieldParamError<T> extends Error {
+    private readonly propInfo: PropValidationInfo<T>;
+
+    constructor(propInfo: PropValidationInfo<T>, expectedType: string) {
+        super(
+            `Warning: Invalid prop "${propInfo.name}" of type "${typeof propInfo.value}" supplied to "${propInfo
```

Comment:
Why this goes into `super`?

Reply:
to leverage the `Error` class methods/properties

Reply:
@freddyucv thoughts?

Reply:
this is ok, you allways have to call the super constructor